src/wolfkrow/core/tasks/task.py

```diff
@@ -365,102 +366,137 @@ def sub_space_for_underscore(string):
 
+        Args:
+            job_name (str): The name of the job that this task is a part of.
+                Only used to generate the script's name.
+            temp_dir (str): Temp directory to write a BASH script to
+            deadline (bool): Whether to prepare this task for Deadline.
+        """
         if self.command_line_executable is None:
-            raise TaskException("WOLFKROW_DEFAULT_COMMAND_LINE_EXECUTABLE variable undefined and no executable specified.")
+            raise TaskException(
+                "WOLFKROW_DEFAULT_COMMAND_LINE_EXECUTABLE variable undefined "
+                "and no executable specified."
+            )
 
         if not self.temp_dir:
             self.temp_dir = temp_dir
 
-        arg_str = ""
+        task_args_dict = {}
 
-        for attribute_name, attribute_obj  in list(self.task_attributes.items()):
+        for attribute_name, attribute_obj in list(self.task_attributes.items()):
             if attribute_obj.serialize is False:
                 continue
 
             value = attribute_obj.__get__(self)
             if value is None or value == attribute_obj.default_value:
                 continue
 
-            value = self._command_line_sanitize_attribute(attribute_name, value, deadline=deadline)
+            sanitised_value = self._command_line_sanitize_attribute(
+                attribute_name, value, deadline=deadline
+            )
+
+            task_args_dict[attribute_name] = sanitised_value
+
+        task_args = []
 
-            # Now put together the arg string
-            arg_str = "{arg_str} --{attribute_name} {value}".format(
-                arg_str=arg_str,
-                attribute_name=attribute_name,
-                value=value
+        if os.path.basename(self.command_line_executable).startswith("wolfkrow_run_task"):
+            # If the executable is Wolfkrow, then write all the args to a JSON
+            # file and pass the path in as a single arg
+            json_file_path = self._get_script_path(
+                extension="json", job_name=job_name, temp_dir=temp_dir
             )
 
+            try:
+                with open(json_file_path, "w") as json_file:
+                    json.dump(task_args_dict, json_file, ensure_ascii=False, indent=4)
+
+            except Exception as exception:
+                raise TaskException(
+                    "Couldn't write args JSON file to path: %s - %s"
+                    % (json_file_path, exception)
+                )
+
+            start_frame = task_args_dict.get("start_frame")
+            end_frame = task_args_dict.get("end_frame")
+
+            # Include the start + end frames, as we want Deadline to be able to
+            # replace them for chunked jobs
+            if "start_frame" not in (None, "None"):
+                task_args.append("--start_frame \"%s\"" % start_frame)
+            if "end_frame" not in (None, "None"):
+                task_args.append("--end_frame \"%s\"" % end_frame)
```

Comment:
Actually... one without the other could be fine for some task definitions... Lets not force anything in here, and instead let the Task's handle this case.

BUT these conditions are currently comparing hard-coded strings rather than the variable... So will always be true. Still need to fix this.